hoomd/hpmc/compute.py

```diff
@@ -7,123 +7,111 @@
 
     Examples::
 
-        mc = hpmc.integrate.sphere(seed=415236)
-        compute.free_volume(mc=mc, seed=123, test_type='B', nsample=1000)
-        log = analyze.log(quantities=['hpmc_free_volume'], period=100, filename='log.dat', overwrite=True)
+        mc = hoomd.hpmc.integrate.Sphere()
+        mc.shape["A"] = {'diameter': 1.0}
+        mc.shape["B"] = {'diameter': 0.2}
+        mc.depletant_fugacity["B"] = 1.5
+        fv = hoomd.hpmc.compute.FreeVolume(mc=mc, test_type='B', nsample=1000)
 
     """
-    def __init__(self, mc, seed, suffix='', test_type=None, nsample=None):
-
-        # initialize base class
-        _compute.__init__(self);
-
-        # create the c++ mirror class
-        cl = _hoomd.CellList(hoomd.context.current.system_definition);
-        hoomd.context.current.system.addCompute(cl, "auto_cl3")
-
-        cls = None;
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            if isinstance(mc, integrate.sphere):
-                cls = _hpmc.ComputeFreeVolumeSphere;
-            elif isinstance(mc, integrate.convex_polygon):
-                cls = _hpmc.ComputeFreeVolumeConvexPolygon;
-            elif isinstance(mc, integrate.simple_polygon):
-                cls = _hpmc.ComputeFreeVolumeSimplePolygon;
-            elif isinstance(mc, integrate.convex_polyhedron):
-                cls = _hpmc.ComputeFreeVolumeConvexPolyhedron;
-            elif isinstance(mc, integrate.convex_spheropolyhedron):
-                cls = _hpmc.ComputeFreeVolumeSpheropolyhedron;
-            elif isinstance(mc, integrate.ellipsoid):
-                cls = _hpmc.ComputeFreeVolumeEllipsoid;
-            elif isinstance(mc, integrate.convex_spheropolygon):
-                cls =_hpmc.ComputeFreeVolumeSpheropolygon;
-            elif isinstance(mc, integrate.faceted_ellipsoid):
-                cls =_hpmc.ComputeFreeVolumeFacetedEllipsoid;
-            elif isinstance(mc, integrate.polyhedron):
-                cls =_hpmc.ComputeFreeVolumePolyhedron;
-            elif isinstance(mc, integrate.sphinx):
-                cls =_hpmc.ComputeFreeVolumeSphinx;
-            elif isinstance(mc, integrate.convex_spheropolyhedron_union):
-                cls = _hpmc.ComputeFreeVolumeConvexPolyhedronUnion
-            elif isinstance(mc, integrate.faceted_ellipsoid_union):
-                cls = _hpmc.ComputeFreeVolumeFacetedEllipsoidUnion
-            elif isinstance(mc, integrate.sphere_union):
-                cls = _hpmc.ComputeFreeVolumeSphereUnion;
-            else:
-                hoomd.context.current.device.cpp_msg.error("compute.free_volume: Unsupported integrator.\n");
-                raise RuntimeError("Error initializing compute.free_volume");
+    def __init__(self, mc, test_type=None, nsample=None):
+        # store metadata
+        param_dict = ParameterDict(
+            mc=integrate.HPMCIntegrator,
+            test_particle_type=OnlyTypes((str, int)),
```

Comment:
```suggestion
            test_particle_type=OnlyTypes(str),
```
Particle types should be strings in user-facing code.